### PR TITLE
feat(elasticsearch): add support for `opensearch-py`

### PR DESCRIPTION
**Pull request by luislew**
## Description

This PR updates the `elasticsearch` integration to add support for `opensearch-py`. This is an alternative to #4611, which implements `opensearch` support as a standalone integration. 

## Checklist
- [x] Followed the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) when writing a release note.
- [x] Add additional sections for `feat` and `fix` pull requests.
- [x] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Motivation

After testing out the solution presented in #4611, I noticed some differences in the Datadog APM UI between the current `elasticsearch` spans and the new `opensearch` spans:
1. `elasticsearch` spans use the `elasticsearch` span type, which is aliased as a DB service within Datadog
2. `elasticsearch.body` span tags have query values replaced with `?`, and the JSON is pretty printed in its own collapsible text box

Until Datadog creates similar mappings for `opensearch` spans, there's utility in preserving the existing `elasticsearch` span names / tags when wrapping `opensearchpy`.

## Design 
As opposed to #4611, the plan here was to preserve all of the existing `elasticsearch` integration and just add `opensearchpy` to the list of modules returned in `_es_modules`. 

## Testing strategy

I moved the OpenSearch integration test into the `elasticsearch` test module and ran the tests locally. We've also been using the following code to patch `opensearchpy` in our application:

```
import opensearchpy
from ddtrace import config
from ddtrace.contrib.elasticsearch.patch import _patch

config.elasticsearch["service"] = "marketplace-opensearch"

_patch(opensearchpy)
```

Screenshot 2022-11-29 at 1 22 53 PM

## Relevant issue(s)

Closes [#3331](https://github.com/DataDog/dd-trace-py/issues/3331)

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Comment by luislew** [comment]
@majorgreys Tagging you here on the simpler alternative to #4611

**Comment by mabdinur** [comment]
Hey @luislew,

Thanks for your contribution. I updated the opensearch tests to use the existing elasticsearch tests. This was done to reduce code duplication. I also updated our test configurations to run the `opensearch` docker service in circleci. Feel free to revert any of my changes if they are not appropriate. 

Also I expect a few opensearch tests to fail. I believe a some versions of opensearch are not fully compatible with the elasticsearch api. We might need to override one or two of the tests in `ElasticsearchPatchTest`. Once those failures are addressed I'll give this PR a final review. 

Thanks again for pushing this forward!

**Review by mabdinur — COMMENTED** [review]
Hey @luislew,

Thanks for your contribution. I updated the opensearch tests to use the existing elasticsearch tests. This was done to reduce code duplication. I also updated our test configurations to run the opensearch docker service in circleci. Feel free to revert any of my changes if they are not appropriate.

There is one failing opensearch test due to some versions of opensearch not being fully compatible with the elasticsearch api. Opensearch v1.1.1 introduced a [breaking change](https://github.com/opensearch-project/opensearch-py/pull/151/files) which removed the `doc_type` arg from `Opensearch.index()`. This test failure needs to be addressed (ex: we can override this test and avoid passing the doctype arg) and we most likely need additional tests/investigation to ensure Opensearch client operations are supported by the existing elasticsearch integration.

Thanks again for pushing this forward!

**Review by mabdinur — DISMISSED** [review]

File: releasenotes/notes/add-opensearch-support-6cc83636317df1a7.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    elasticsearch: Update `elasticsearch` integration to add support for ``opensearch-py``. See `the elasticsearch documentation <https://ddtrace.readthedocs.io/en/stable/integrations.html#elasticsearch>`_
```
Comment (brettlangdon) [inline releasenotes/notes/add-opensearch-support-6cc83636317df1a7.yaml:4]:
```suggestion
    elasticsearch: Update ``elasticsearch`` integration to add support for ``opensearch-py``. See `the elasticsearch documentation <https://ddtrace.readthedocs.io/en/stable/integrations.html#elasticsearch>`_
```

**Review by brettlangdon — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Comment by luislew** [comment]
@brettlangdon I see that merging is blocked because the "base branch requires all commits to be signed". Do I need to take any action to fix this, or can you resolve?

**Pull request merged**